src/store/enterprise/managedStorage.ts

```diff
@@ -15,29 +15,54 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+/**
+ * @file A wrapper around the browser.storage.managed that tries to smooth over its initialization quirks and provide
+ * an interface for React's useExternalStore
+ */
+
 import { type ManagedStorageState } from "@/store/enterprise/managedStorageTypes";
 import { isEmpty, once } from "lodash";
 import { expectContext } from "@/utils/expectContext";
 import pMemoize, { pMemoizeClear } from "p-memoize";
 import { pollUntilTruthy } from "@/utils/promiseUtils";
+import type { Nullishable } from "@/utils/nullishUtils";
 
-const MAX_MANAGED_STORAGE_WAIT_MILLIS = 2000;
+// 2024-02-15: bumped to 4.5s because 2s was too short: https://github.com/pixiebrix/pixiebrix-extension/issues/7618
+//   Privacy Badger uses 4.5s timeout, but thinks policy should generally be available within 2.5s. In installer.ts,
+//   we check for app tabs to skip the linking wait if the user appears to be installing from the web store.
```

Comment:
nit/suggestion to clarify here:
```suggestion
//   skip waiting for managed storage before linking the Extension if the user appears to be installing from the web store.
```